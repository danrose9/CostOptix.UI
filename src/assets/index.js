export { default as AZURE96 } from './azure-96.png';
export { default as AWS96 } from './amazon-web-services-96.png';
export { default as MICROSOFT96 } from './microsoft-96.png';
export { default as SALESFORCE96 } from './salesforce-96.png';
export { default as GOOGLE96 } from './google-cloud-96.png';
export { default as LOGOBLUE } from './logo-blue.png';
export { default as LOGOWHITE } from './logo-white.png';
export { default as ERROR404 } from './404-error.png';
export { default as BROKENPAGE } from './broken-page.png';
export { default as HOME_BACKGROUND } from './home-background.png';
export { default as CHECKLIST } from './checklist.png';
export { default as SUPPORT } from './support.png';
