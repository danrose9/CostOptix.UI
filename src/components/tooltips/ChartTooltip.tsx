import React from 'react';
import { Card } from 'semantic-ui-react';
import { ProviderImage } from '../ProviderImage';

interface IChartTooltipProps {
  instance?: { name: string; value: number }[];
  label?: string;
  symbol?: string;
}

export const ChartTooltip: React.FC<IChartTooltipProps> = ({ instance, label, symbol }) => {
  return (
    <Card raised>
      {instance?.map((provider, index) => {
        return (
          <Card.Content key={index} extra>
            <ProviderImage floated="right" size="mini" provider={provider.name} />
            <Card.Header>{provider.name}</Card.Header>
            <Card.Meta>
              {symbol}
              {provider.value}
            </Card.Meta>
          </Card.Content>
        );
      })}
      <Card.Content textAlign="right">
        <Card.Description>{label}</Card.Description>
      </Card.Content>
    </Card>
  );
};
export default ChartTooltip;
