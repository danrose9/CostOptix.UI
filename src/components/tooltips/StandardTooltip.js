import React, { Component } from 'react';
import { Card } from 'semantic-ui-react';
import { ProviderImage } from '../ProviderImage';

export default class StandardTooltip extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }
  render() {
    const { provider, accountName } = this.props.instance;

    return (
      <Card>
        <Card.Content>
          <ProviderImage floated="right" size="mini" provider={provider} />
          <Card.Header>{provider}</Card.Header>
          <Card.Meta>{accountName}</Card.Meta>
        </Card.Content>

        <Card.Content>{this.props.children}</Card.Content>
      </Card>
    );
  }
}
