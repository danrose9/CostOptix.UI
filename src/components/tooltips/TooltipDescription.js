import React, { Fragment } from 'react';
import { Card, Divider } from 'semantic-ui-react';

export const TooltipDescription = (instance) => {
  const { service, resourceName } = instance.instance;
  return (
    <Fragment>
      <Card.Description>
        <strong>Service</strong> : {service}
      </Card.Description>
      <Card.Description>
        <strong>Resource</strong> : {resourceName}
      </Card.Description>
      <Divider />
    </Fragment>
  );
};
export default TooltipDescription;
