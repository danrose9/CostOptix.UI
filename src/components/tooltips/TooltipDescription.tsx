import React, { Fragment } from 'react';
import { Card, Divider } from 'semantic-ui-react';
import { getEmbeddedResourceName } from 'src/utils/stringFormatter';

interface ITooltipDescription {
  instance: {
    service: string;
    resourceName: string;
  };
}

const TooltipDescription: React.FC<ITooltipDescription> = ({ instance }) => {
  const { service, resourceName } = instance;

  return (
    <Fragment>
      <Card.Description>
        <strong>Service</strong> : {getEmbeddedResourceName(service)}
      </Card.Description>
      <Card.Description>
        <strong>Resource</strong> : {getEmbeddedResourceName(resourceName)}
      </Card.Description>
      <Divider />
    </Fragment>
  );
};
export default TooltipDescription;
