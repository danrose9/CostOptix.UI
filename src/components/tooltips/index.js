export { default as StandardTooltip } from './StandardTooltip';
export { default as TooltipDescription } from './TooltipDescription';
export { default as ChartTooltip } from './ChartTooltip';
