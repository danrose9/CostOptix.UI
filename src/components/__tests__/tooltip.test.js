import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { StandardTooltip, TooltipDescription, ChartTooltip } from '../tooltips/index';

import renderer from 'react-test-renderer';
import { AWS96 } from '../../assets';

afterEach(() => {
  cleanup();
});

const instance = [
  { name: 'AWS', value: 456.1 },
  { name: 'Azure', value: 54.2 },
  { name: 'Microsoft', value: 353.3 },
];

describe('Tooltips', () => {
  const TestComponent = () => {
    return (
      <StandardTooltip instance={{ provider: 'Provider', accountName: 'Account Name' }}>
        <TooltipDescription instance={{ service: 'Service', resourceName: 'Resource Name' }} />
      </StandardTooltip>
    );
  };
  test('should render Standard Tooltip', () => {
    render(<TestComponent />);
  });

  test('should render Chart Tooltip', () => {
    render(<ChartTooltip instance={instance} label="label" symbol="$" />);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<TestComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
