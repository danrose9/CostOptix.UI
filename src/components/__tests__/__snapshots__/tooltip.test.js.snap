// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltips matches snapshot 1`] = `
<div
  className="ui card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <img
      alt="ServiceProviderLogo"
      className="ui mini right floated image"
      style={
        Object {
          "margin": "0.2em",
        }
      }
    />
    <div
      className="header"
    >
      Provider
    </div>
    <div
      className="meta"
    >
      Account Name
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="description"
    >
      <strong>
        Service
      </strong>
       : 
      Service
    </div>
    <div
      className="description"
    >
      <strong>
        Resource
      </strong>
       : 
      Resource Name
    </div>
    <div
      className="ui divider"
    />
  </div>
</div>
`;
